### PR TITLE
cusp: eliminate global dependence on subclass state

### PR DESCRIPTION
**Pull request by tylershunt**
These `*Impl` functions return data that doesn't change once partitioner
is initialized. Changing them to be memoized instead means that the
state of `DistGraph` accessible through public members does does not
depend on the state of `NewGenericGraph` or `MiningGraph`.

Depends on #337; only the last commit will be in the final merge.

Signed-off-by: Tyler Hunt <thunt@katanagraph.com>

### CODE REVIEW

File: libcusp/include/galois/graphs/DistributedGraph.h [hunk 1]
```diff
@@ -323,15 +318,18 @@ class DistGraph {
...
-      num_global_nodes_; //!< Total nodes in the global unpartitioned graph.
-  uint64_t
-      num_global_edges_; //!< Total edges in the global unpartitioned graph.
-  uint32_t num_nodes_;   //!< Num nodes in this graph in total
-  uint64_t num_edges_;   //!< Num edges in this graph in total
+  uint64_t num_global_nodes_{
+      0}; //!< Total nodes in the global unpartitioned graph.
+  uint64_t num_global_edges_{
+      0};              //!< Total edges in the global unpartitioned graph.
+  uint32_t num_nodes_; //!< Num nodes in this graph in total
```
Comment (gurbinder533) [inline libcusp/include/galois/graphs/DistributedGraph.h:331]: Initialize num_nodes_ and num_edges_ as well?

Comment (gurbinder533) [reply libcusp/include/galois/graphs/DistributedGraph.h:331]: These are initialized in the constructor, so nevermind.

Comment (tylershunt) [reply libcusp/include/galois/graphs/DistributedGraph.h:328]: Sure; I just moved these up out of the constructor but I'm happy to add that as well.

**Comment by tylershunt** [comment]
I also removed dead functions `getHostID` and `isOwned`.

**Comment by l-hoang** [comment]
> I also removed dead functions `getHostID` and `isOwned`.

Were they actually dead functions? I seem to recall that they were used somewhere in the code unless the uses have been removed.

**Comment by gurbinder533** [comment]
> > I also removed dead functions `getHostID` and `isOwned`.
> 
> Were they actually dead functions? I seem to recall that they were used somewhere in the code unless the uses have been removed.

Yeah, I thought so too. But I can't seem to find any reference now and they are not very complex to re-introduce if we need them.

**Comment by tylershunt** [comment]
> Were they actually dead functions?

Doing a grep of `lib*` and `lonestar` the only instance of `getHostID()` belong to `galois::runtime` and I can't find the string`isOwner` at all (after this change).

**Pull request merged**